projects/plugins/boost/app/assets/src/js/features/speed-score/pop-out/pop-out.tsx

```diff
@@ -0,0 +1,154 @@
+			<p>
+				{ __(
+					'Most of the time Jetpack Boost will increase your site speed, but there may be cases where your score does not increase.',
+					'jetpack-boost'
+				) }
+			</p>
+			<p>
+				{ __(
+					'Try refreshing your score, and if it doesn’t help, check our guide on improving your site speed score:',
+					'jetpack-boost'
+				) }
+			</p>
+		</>
+	),
+	cta: __( 'Read the guide', 'jetpack-boost' ),
+	ctaLink: 'https://jetpack.com/support/speed-up-your-site/',
+};
+
+/**
+ * Wrapper for PopOut which provides a data sync context. Can be removed once we know the
+ * parent of PopOut is always wrapped in a DataSyncProvider.
+ *
+ * @param {Props} props Properties.
+ */
+export default function PopOut( props: Props ) {
+	return (
+		<DataSyncProvider>
+			<_PopOut { ...props } />
+		</DataSyncProvider>
+	);
+}
+
+/**
+ * Helper hook - Use data sync to track which score popouts have been dismissed.
+ *
+ * @param id - The id of the score popout to track. Accepts null (which will result in dummy return values) for ease of use.
+ */
+function useDismissedScoreAlerts( id: string | null ): [ boolean, () => void ] {
+	const [ { data }, { mutate } ] = useDataSync(
+		'jetpack_boost_ds',
+		'dismissed_score_prompt',
+		z.array( z.string() )
+	);
+
+	// If no id provided, give back a dummy function with "yes, dismissed".
+	if ( ! id ) {
+		return [ true, () => {} ];
+	}
+
+	const isDismissed = data?.includes( id ) || false;
+	const dismiss = () => {
+		if ( ! isDismissed ) {
+			mutate( [ ...( data || [] ), id ] );
+		}
+	};
+
+	return [ isDismissed, dismiss ];
+}
+
+function _PopOut( { scoreChange, onClose }: Props ) {
+	const [ message, setMessage ] = useState< ScoreChangeMessage | null >( null );
+	const hasScoreChanged = scoreChange !== false && Math.abs( scoreChange ) > 5;
+
+	// Keep the last score change message in state, so we can animate it out after the user dismisses it.
+	useEffect( () => {
+		if ( hasScoreChanged ) {
+			setMessage( scoreChange > 0 ? fasterMessage : slowerMessage );
+		}
+	}, [ hasScoreChanged, scoreChange ] );
+
+	// Use datasync to track which score alerts have been dismissed.
+	const [ isDismissed, dismiss ] = useDismissedScoreAlerts( message?.id );
+
+	// Callback to handle the user dismissing the message by clicking the CTA or the dismiss option
+	function handleDismiss() {
+		dismiss();
+		onClose();
+	}
+
+	return (
+		<div id="parent" className={ styles.wrapper }>
+			<CSSTransition
+				in={ hasScoreChanged && ! isDismissed }
+				appear={ true }
+				classNames={ { ...styles } }
+			>
+				<div className={ styles.card }>
+					<CloseButton onClick={ onClose } />
+
+					<h3 className={ styles.headline }>{ message && message.title }</h3>
+
+					<p>{ message && message.body }</p>
+
+					<a
+						className="jb-button--primary"
+						href={ message && message.ctaLink }
+						target="_blank"
+						rel="noreferrer"
+						onClick={ handleDismiss }
+					>
+						{ message && message.cta }
+					</a>
```

Comment:
This can also be a `Button`. For full usage info, I recommend running storybook and seeing the example.

Reply:
From an accessibility perspective, it's a link. You click it to open a webpage.

It's kind of funny. We have a link to a website, which we deliberately style to look like a button, and a button which we deliberately style to look like a link.

But functionally, the button-which-looks-like-a-link is used to interact with the page you're on - it's a control for what the site should show you. The link-that-looks-like-a-button takes you to an external website.